src/pyodide/internal/setupPackages.ts

```diff
@@ -189,19 +189,26 @@ export function getSitePackagesPath(Module: Module): string {
  * details, so even though we want these directories to be on sys.path, we
  * handle that separately in adjustSysPath.
  */
-export function mountLib(Module: Module, info: TarFSInfo): void {
+export function mountSitePackages(Module: Module, info: TarFSInfo): void {
   const tarFS = createTarFS(Module);
-  const mdFS = createMetadataFS(Module);
   const site_packages = getSitePackagesPath(Module);
   Module.FS.mkdirTree(site_packages);
-  Module.FS.mkdirTree('/session/metadata');
   if (!LOAD_WHEELS_FROM_R2 && !LOAD_WHEELS_FROM_ARTIFACT_BUNDLER) {
     // if we are not loading additional wheels, then we're done
     // with site-packages and we can mount it here. Otherwise, we must mount it in
     // loadPackages().
     Module.FS.mount(tarFS, { info }, site_packages);
   }
+}
+
+export function mountWorkerFiles(Module: Module) {
+  Module.FS.mkdirTree('/session/metadata');
+  const mdFS = createMetadataFS(Module);
   Module.FS.mount(mdFS, {}, '/session/metadata');
+  simpleRunPython(
+    Module,
+    `import sys; sys.path.append("/session/metadata"); del sys`
```

Comment:
I might be totally off base here but wouldn't we want `sys.path.insert(0, ...)` here?
If a user has http.py file with his main.py file we'd want his http.py file to take precedence over the builtin http module when he uses `import http` no?

Reply:
Edit: I can see this was moved so obviously it isn't specific to this PR but just a general question

Reply:
Actually, and sorry for being annoying but the move actually does change behaviour where now site-packages are before session/metadata in path in contrast to previous behaviour.

Reply:
Good point I put that back. Good idea about `sys.path.insert(0, ...)` I think that is right. I'll make that change in a separate PR though.